services/QuillLMS/client/app/bundles/Teacher/components/classrooms/active_classrooms.tsx

```diff
@@ -1,737 +1,600 @@
-export const cleverClassroomsEmptyModal = 'cleverClassroomsEmptyModal'
-export const googleClassroomsEmptyModal = 'googleClassroomsEmptyModal'
+export const reauthorizeProviderModal = 'reauthorizeProviderModal'
+export const noClassroomsToImportModal = 'noClassroomsToImportModal'
 export const viewAsStudentModal = 'viewAsStudentModal'
 
-export default class ActiveClassrooms extends React.Component<ActiveClassroomsProps, ActiveClassroomsState> {
-  constructor(props) {
-    super(props)
+const bookEmptySrc = `${process.env.CDN_URL}/images/illustrations/book-empty.svg`
+const canvasIconSrc = `${process.env.CDN_URL}/images/icons/canvas.svg`
+const cleverIconSrc = `${process.env.CDN_URL}/images/icons/clever.svg`
+const googleClassroomIconSrc = `${process.env.CDN_URL}/images/icons/google-classroom.svg`
+const reorderSrc = `${process.env.CDN_URL}/images/icons/reorder.svg`
+
+interface ActiveClassroomsProps {
+  canvasLink: string
+  classrooms: Array<any>
+  cleverLink: string
+  googleLink: string
+  coteacherInvitations: Array<any>
+  user: any
+}
 
-    this.state = {
-      showSnackbar: false,
-      classrooms: props.classrooms.filter(classroom => classroom.visible),
-      coteacherInvitations: props.coteacherInvitations,
-      cleverClassrooms: [],
-      googleClassrooms: []
+const ActiveClassrooms = ({
+  canvasLink,
+  classrooms: initialClassrooms,
+  cleverLink,
+  coteacherInvitations: initialCoteacherInvitations,
+  googleLink,
+  user,
+}: ActiveClassroomsProps) => {
+  const { provider } = user
+  const providerClassName= providerLookup[provider].className
+  const retrieveClassroomsPath = `/${providerClassName}_integration/teachers/retrieve_classrooms`
+  const retrieveClassroomsEventName = `${providerClassName}-classrooms-retrieved`
+
+  const isCanvasUser = provider === canvasProvider;
+  const isCleverUser = provider === cleverProvider;
+  const isGoogleUser = provider === googleProvider;
+
+  const reauthorizeLink = {
+    [canvasProvider]: canvasLink,
+    [cleverProvider]: cleverLink,
+    [googleProvider]: googleLink,
+  }[provider]
+
+  const [classrooms, setClassrooms] = useState(initialClassrooms)
+  const [coteacherInvitations, setCoteacherInvitations] = useState(initialCoteacherInvitations)
+  const [providerClassrooms, setProviderClassrooms] = useState([])
+  const [providerClassroomsLoading, setProviderClassroomsLoading] = useState(false)
+  const [pendingImportFromProviderRequest, setPendingImportFromProviderRequest] = useState(false)
+
+  const [isSnackbarVisible, setIsSnackbarVisible] = useState(false)
+  const [snackbarCopy, setSnackbarCopy] = useState('')
+  const [visibleModal, setVisibleModal] = useState(null)
+  const [selectedClassroomId, setSelectedClassroomId] = useState(null)
+
+  const providerIconSrc = {
+    [canvasProvider]: canvasIconSrc,
+    [cleverProvider]: cleverIconSrc,
+    [googleProvider]: googleClassroomIconSrc,
+  }
+
+  useEffect(() => {
```

Comment:
There's actually a `useSnackbarMonitor` hook that you can use in functional components instead of having to write custom logic. If you search the codebase for that, you'll see examples of the implementation.

Reply:
Cool!  Thank you for the heads up.